### DISCUSSION TITLE
Code

### DISCUSSION THREAD

**Discussion by wartburggraaf** [nl-design-system/backlog#327] [Component suggestions]
<img width="480" alt="visuele weergave van een Code component" src="https://github.com/user-attachments/assets/16fff079-b1a4-4857-83b8-1a1c7e8ac968">

We willen graag meer leren over de Code component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat de component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1725369265485599)

## Naam

Code

## Doel

Computercode die onderdeel is van lopende tekst.

## Gerelateerde componenten

[Code Block](https://github.com/orgs/nl-design-system/discussions/328)

## Richtlijnen

Acceptatiecriteria: [NL Design System Acceptatiecriteria voor Code](https://nldesignsystem.nl/code)

## Onderzoek

**Comment by nies333** [comment]
@ninya9k hi did u remember me what happened to ur site

**Comment by jeffreylauwers** [comment]
## Utrecht

- [Storybook](https://nl-design-system.github.io/utrecht/storybook/?path=/docs/css_css-code--docs)
- [GitHub](https://github.com/nl-design-system/utrecht/tree/main/components/code)

---

### Screenshots / Varianten

**Storybook**

Default
Markup with the `<code>` element. Styling via `utrecht-code` class name.
![Scherm­afbeelding 2024-09-04 om 13 22 05](https://github.com/user-attachments/assets/088b374d-4094-48d3-893b-435c19314452)

Ligatures in code
“fi,” “ff,” en “fl“ zijn veelvoorkomende [ligaturen](https://nl-design-system.github.io/utrecht/storybook-css/%3Chttps://nl.wikipedia.org/wiki/Ligatuur_(typografie)%3E). In code horen ze getoond te worden als twee karakters. Sommige ligaturen zijn beschikbaar in Unicode als een enkel karakter, zoals “ﬁ” (U+FB01), Ĳ (U+0132) en ĳ (U+0133).
![Scherm­afbeelding 2024-09-04 om 13 22 17](https://github.com/user-attachments/assets/97eb02b0-f491-4fe6-856e-15f24f0ac7ba)

Code in link
De code die een link is moet herkenbaar zijn als link.
![Scherm­afbeelding 2024-09-04 om 13 22 22](https://github.com/user-attachments/assets/c8a884f2-2696-42c1-acc8-5248b9186f5e)

Code in a paragraph
Lopende tekst met code moet goed leesbaar zijn, de code moet niet zo afleidend opgemaakt zijn dat de tekst slecht leesbaar wordt.
![Scherm­afbeelding 2024-09-04 om 13 22 27](https://github.com/user-attachments/assets/4c5c9abc-11b9-4557-a230-c93bb49c0986)

Code in a heading 1
De lettergrootte van de code moet aanpassen aan de parent.
![Scherm­afbeelding 2024-09-04 om 13 22 40](https://github.com/user-attachments/assets/d095b81a-dda3-4904-a337-2a8248f49436)

Code in a heading 2
De lettergrootte van de code moet aanpassen aan de parent.
![Scherm­afbeelding 2024-09-04 om 13 22 50](https://github.com/user-attachments/assets/dee76d9f-fffa-4492-9cc9-5f65ab682451)

---

### Notities

- Documenatie beschikbaar (Storybook)
- Veel verschillende varianten beschikbaar. Ook in combinatie met headings (nu enkel H1 en H2 meegenomen in screenshots).
- Via 'Element' property kan er gekozen worden voor `code` of `span`.
- Huidige implementatie heeft geen achtergrond kleur, maar er is wel een design token voor beschikbaar.

**Comment by jeffreylauwers** [comment]
## Atlassian

[Documentatie website](https://atlassian.design/components/code/examples)

---

### Screenshots / Varianten

**Documentatie website**

Inline code
Formatted code can appear in a variety of contexts, increasing the legibility and contrasting it against default paragraph text.
Use inline code when you wish to highlight a short code snippet from the surrounding default text, such as when referencing variable names.

![Scherm­afbeelding 2024-09-04 om 13 35 20](https://github.com/user-attachments/assets/517e8f41-a5c0-4ed3-b09c-4216403d9d8b)

---

### Notities

- Onder de naam 'Inline code'
- Lichtgrijze, transparent, achtergrond kleur. Subtiel afgeronde hoekjes.

**Comment by jeffreylauwers** [comment]
## IBM

[Documentatie website](https://carbondesignsystem.com/components/code-snippet/usage/)

---

### Screenshots / Varianten

**Documentatie website**

Code snippet - 'Inline' variant
Code snippets are strings or small blocks of reusable code that can be copied and inserted in a code file.

![Scherm­afbeelding 2024-09-04 om 13 39 04](https://github.com/user-attachments/assets/96e89691-762a-46a7-bce6-08c6a20244ae)

---

### Notities

- Onder de naam 'Code snippet' als variant 'Inline'.
- Lichtgrijze achtergrond kleur. Subtiel afgeronde hoekjes.
- Door te klikken kopieer je de code.

**Comment by jeffreylauwers** [comment]
## GitHub

[Documentatie website](https://primer.style/foundations/typography#code)

---

### Screenshots / Varianten

**Documentatie website**

![Scherm­afbeelding 2024-09-04 om 13 49 11](https://github.com/user-attachments/assets/4754f59c-3c93-442d-8d63-3eacbed06232)

---

### Notities

- Onder de naam 'Inline code block' beschikbaar als typografisch element
- Zie ook de code weergave bij 'Propery' en 'Token' in het screenshot. Hier wordt code net even anders weergegeven en dit is niet terug te vinden als component. Bij 'Token' is er een kopieer optie aan toegevoegd.

**Comment by jeffreylauwers** [comment]
## NL Design System

[Documentatie website](https://www.nldesignsystem.nl/richtlijnen/formulieren/descriptions/koppelen)

---

### Screenshots / Varianten

**Documentatie website**

Op de documentatie website van NL Design Sytem gebruiken we de Code component van Utrecht.

![Scherm­afbeelding 2024-09-04 om 13 42 43](https://github.com/user-attachments/assets/f5d972dd-c5fa-48a2-a856-aab861936569)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Code op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=code+%23326)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Community ✨

Help je mee hem door de Candidate stappen te krijgen?
[Code op het Candidate bord](https://github.com/orgs/nl-design-system/projects/32/views/1?filterQuery=%23326)

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Candidate ✨

Help je mee hem door de Hall of Fame stappen te krijgen?
[Code op het Hall of Fame bord](https://github.com/orgs/nl-design-system/projects/30/views/1?filterQuery=code+326)
